sdk/messaging/eventgrid/azsystemevents/error.go

```diff
@@ -0,0 +1,20 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azsystemevents
+
+// Error is an error that is included as part of a system event.
+type Error struct {
+	// Code is an error code from the service producing the system event.
+	Code string
+
+	message string
```

Comment:
Debating if this should be exported for fakes purposes?

Reply:
@JeffreyRichter and I had a bit of a back and forth on this. I totally don't mind it being exported. Jeff, what do you think?

Reply:
I don't know enough about how fakes work with AMQP.  To me, this is like how we only expose HTTP status code and code from ResponseError but it holds on to the raw JSON error response body. This has a way of working with fakes, right? If so, I'd expect this to work the same way.

Reply:
Just to be clear, this isn't AMQP, it's protocol independent (the events can be received from anywhere). 

@jhendrixMSFT , thinking about this more this is the 'message' part, which is considered the cosmetic part. They can already change Code, which is really the only piece they should use.

So maybe we leave this one unexported.

Reply:
That's fine, we can always export it later if required.